exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/DataTypeProvider.kt

```diff
@@ -148,7 +148,7 @@ abstract class DataTypeProvider {
         }
 
         e is LiteralOp<*> -> (e.columnType as IColumnType<Any?>).valueAsDefaultString(e.value)
-        e is Function<*> -> "$e"
+        e is Function<*> -> if (currentDialect is SQLiteDialect) "($e)" else "$e"
```

Comment:
In SQLite, complex functions need to be wrapped in parentheses. Since there is no way to know if an expression is a complex function or not, I decided to wrap all functions in parentheses for SQLite.

Reply:
Could you confirm what I'm maybe missing? What changes in this PR will affect SQLite that require this? The tests seem to pass without this change.

This is the purpose of `ComplexExpression`, isn't it? To wrap any expression with [that implements this marker](https://github.com/JetBrains/Exposed/blob/main/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Op.kt#L789) with parentheses.

Reply:
`testDefaultExpressions01` would fail for SQLite without this change. This is a separate issue that was discovered while working on the issue for this PR. We get a syntax error when we use a complex expression (`AbsFunction` in this case) as a default in SQLite because Exposed is not wrapping it with parentheses. I will open a separate PR to fix this.